pyobis/obisutils.py

```diff
@@ -17,39 +17,22 @@ class NoResultException(Exception):
 
 
 def build_api_url(url, args):
-    return url + "?" + urlencode({k: v for k, v in args.items() if v is not None})
-
-
-class OBISQueryResult:
     """
-    Return OBIS API URL
+    Builds the API URL based on the base url and the arguments
     """
-
-    def get_search_url(self):
-        """
-        Get the corresponding API URL for the query.
-
-        :return: OBIS API URL for the corresponding query
-
-        Usage::
-
-            # supports usage for any query from any module
-            from pyobis.taxa import TaxaQuery
-            query = TaxaQuery
-            query.search(scientificname="Mola mola")
-            query.get_search_url()
-
-        """
-        return (
-            self.url
-            + "?"
-            + urlencode({k: v for k, v in self.args.items() if v is not None})
-        )
+    return url + "?" + urlencode({k: v for k, v in args.items() if v is not None})
 
 
 def obis_GET(url, args, ctype, **kwargs):
-    """Handles technical details of sending GET request to the API"""
-    out = requests.get(url, params=args, **kwargs)
+    """
+    Handles technical details of sending GET request to the API
+    """
+    headers = {
+        "User-Agent": "Mozilla/5.0 (X11; CrOS x86_64 12871.102.0) AppleWebKit/537.36 (KHTML, like Gecko)\
```

Comment:
we might consider setting a user-agent string that specifies pyobis instead of this one. I assume this was copied from a browser?

Reply:
Yes, this is a browser user agent string, but I don't know if there are any UA detection tool into OBIS API. If it would be the case then, any other user agent string might work slower than the browser user strings because that might be used to put rate limits on bot networks. 

And, I don't know why but the server response time is too high today for me. This request https://api.obis.org/v3/occurrence?taxonid=2&offset=0&mof=False&size=1 is taking 10.69 s waiting for server response while only 1.63 ms for content download. This is something unusual. I tried clearing DNS Cache, changing IPs, also tried VPNs, but the time taken is same.

Reply:
Maybe this is something related to OBIS performing internal searches for all records and returning only the specified user size. This could be a critical bottleneck before pyobis being performant, but is a backend issue (something as a high-level design problem). I tested `taxonid`s with lesser records footprint and the server wait time was nearly equal to content download time for a similar number of records. I analysed many requests for different number of records and found that the server wait time was a significant portion from 50% to taking nearly 95% of the total time for higher taxons. Therefore I came to the conclusion that for most users, more than network speed, server response time is a bigger bottleneck.